pyronear/datasets/openfire.py

```diff
@@ -101,38 +116,55 @@ def _check_exists(self, train=True):
         else:
             return self._root.joinpath(self._processed, self.test_file).is_file()
 
-    def download(self, threads=None, valid_pct=None):
-        """Download the OpenFire data if it doesn't exist in processed_folder already.
-
+    def create_directories(self, max_files=None):
+        """
+        Create the directory structure to hold the images and training_file/test_file.
+        Download the json file with annotations, add 'fname' with local filenames
+        
         Args:
-            threads (int, optional): Number of threads to use for dataset downloading.
             valid_pct (float, optional): Percentage of training set used for validation.
+            max_files (int, optional): maximum number of files to use. Default: None (all)
         """
-
-        if self._check_exists(train=True) and self._check_exists(train=False):
-            return
-
         self._root.joinpath(self._raw).mkdir(parents=True, exist_ok=True)
         self._root.joinpath(self._processed).mkdir(parents=True, exist_ok=True)
 
         # Download annotations
         download_url(self.url, self._root.joinpath(self._raw), filename=self.url.rpartition('/')[-1], verbose=False)
         with open(self._root.joinpath(self._raw, self.url.rpartition('/')[-1]), 'rb') as f:
-            annotations = json.load(f)
+            try:
+                self.annotations = json.load(f)[:max_files]
+            except TypeError:
+                raise ValueError("max_files must an int between -1 and len(dataset)")
 
-        # Download actual images
-        training_set, test_set = [], []
+        # Add the local filename to the annotations
+        for idx, a in enumerate(self.annotations):
+          a['fname'] = f"{idx:06}.{get_fname(a['url']).rpartition('.')[-1]}"
```

Comment:
Spacing issue here (should be 4 spaces rather than 2)

Reply:
Fixed